storage/data-plane/src/test/java/com/microsoft/azure/storage/blob/HelperTest.groovy

```diff
@@ -99,8 +99,10 @@ class HelperTest extends APISpec {
 
         when:
         def parts = URLParser.parse(cu.createBlobURL(blobName).toURL())
-        parts.withSasQueryParameters(v.generateSASQueryParameters(primaryCreds)).withScheme("https")
-        def bu = new AppendBlobURL(parts.toURL(), StorageURL.createPipeline(new AnonymousCredentials(),
+        SASQueryParameters parameters = v.generateSASQueryParameters(primaryCreds)
+        setSASQueryParametersSignature(parameters)
```

Comment:
Where is this defined? We shouldn't be setting the signature.

Reply:
The signature field is randomly generated. I used the reflection to inject the signature field. In record, it generated the signature as it is (same way as before. I did not change any logic generated signature, just get the signature from parameter, and store in json file). In playback, we will retrieve the signature from json. This is the only way to do the mock request and response without ignoring any field.

Reply:
I see. Can we change the name a bit then? Something like prepareSignatureForRecording or something like that?

Reply:
Done